File: ws/bonilla/u2/ws12Inheritance/FarmSystem/src/main/java/ec/edu/espe/farmsystem/model/FarmAnimal.java
package ec.edu.espe.farmsystem.model;
import java.util.Date;

/**
 *
 * @author Arelis Samantha Bonilla Cruz, @ESPE
 */
public abstract class FarmAnimal {
    private int id;
    private String breed;
    private Date bornOn;
    private String gender;
    private boolean isAbleToReproduce;
    private float weight;
    private Cage cage;
    
    public int getAgeInMonths(){
        //T0D0 compute the age in months
        return 0;
    }

    @Override
    public String toString() {
        return "FarmAnimal{" + "id = " + getId() + ", breed = " + getBreed() + ", bornOn = " + getBornOn() + ", gender = " + getGender() + ", isAbleToReproduce = " + isIsAbleToReproduce() + ", weight = " + getWeight() + ", cage = " + getCage() + '}';
    }
    
    public void assignCage(Cage cage){
        this.setCage(cage);
    }

    public FarmAnimal(int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        this.id = id;
        this.breed = breed;
        this.bornOn = bornOn;
        this.gender = gender;
        this.isAbleToReproduce = isAbleToReproduce;
        this.weight = weight;
        this.cage = cage;
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the breed
     */
    public String getBreed() {
        return breed;
    }

    /**
     * @param breed the breed to set
     */
    public void setBreed(String breed) {
        this.breed = breed;
    }

    /**
     * @return the bornOn
     */
    public Date getBornOn() {
        return bornOn;
    }

    /**
     * @param bornOn the bornOn to set
     */
    public void setBornOn(Date bornOn) {
        this.bornOn = bornOn;
    }

    /**
     * @return the gender
     */
    public String getGender() {
        return gender;
    }

    /**
     * @param gender the gender to set
     */
    public void setGender(String gender) {
        this.gender = gender;
    }

    /**
     * @return the isAbleToReproduce
     */
    public boolean isIsAbleToReproduce() {
        return isAbleToReproduce;
    }

    /**
     * @param isAbleToReproduce the isAbleToReproduce to set
     */
    public void setIsAbleToReproduce(boolean isAbleToReproduce) {
        this.isAbleToReproduce = isAbleToReproduce;
    }

    /**
     * @return the weight
     */
    public float getWeight() {
        return weight;
    }

    /**
     * @param weight the weight to set
     */
    public void setWeight(float weight) {
        this.weight = weight;
    }

    /**
     * @return the cage
     */
    public Cage getCage() {
        return cage;
    }

    /**
     * @param cage the cage to set
     */
    public void setCage(Cage cage) {
        this.cage = cage;
    }
    
    
}

File: hw/aymacana/u2/hw19Inheritance/FarmSystemJS/FarmAnimal.js
class FarmAnimal {
    constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        this._id = id;
        this._breed = breed;
        this._bornOn = bornOn;
        this._gender = gender;
        this._isAbleToReproduce = isAbleToReproduce;
        this._weight = weight;
        this._cage = cage;
    }

    getAgeInMonths() {
        return 0;
    }

    assignCage(cage) {
        this._cage = cage;
    }

    get id() { return this._id; }
    get breed() { return this._breed; }
    get bornOn() { return this._bornOn; }
    get gender() { return this._gender; }
    get isAbleToReproduce() { return this._isAbleToReproduce; }
    get weight() { return this._weight; }
    get cage() { return this._cage; }

    set id(id) { this._id = id; }
    set breed(breed) { this._breed = breed; }
    set bornOn(bornOn) { this._bornOn = bornOn; }
    set gender(gender) { this._gender = gender; }
    set isAbleToReproduce(isAbleToReproduce) { this._isAbleToReproduce = isAbleToReproduce; }
    set weight(weight) { this._weight = weight; }
    set cage(cage) { this._cage = cage; }

    toString() {
        return `FarmAnimal{id=${this._id}, breed='${this._breed}', ` +
               `bornOn=${this._bornOn}, gender='${this._gender}', ` +
               `isAbleToReproduce=${this._isAbleToReproduce}, ` +
               `weight=${this._weight}, cage=${this._cage}}`;
    }
}

module.exports = { FarmAnimal };